src/app/core/services/api.service.ts

```diff
@@ -283,8 +283,19 @@ export class ApiService {
         namespace:"vm.delete",
         args:[],
         responseEvent:"VmDeleted"
-      }
+      },
+      async preProcessor(def:ApiCall, self) {
+        return await self.dialog.confirm("Delete", `Delete ${def.args[1].name} ?`).toPromise().then((res)=>{
```

Comment:
It would be nice to help the user verify what they are deleting (Delete VM {name}), but I don't know if this is a VM-specific routine or just generic.

Reply:
<img width="793" alt="screen shot 2019-01-08 at 10 49 19 am" src="https://user-images.githubusercontent.com/2560355/50841643-29f9b000-1333-11e9-9672-def64c9c303c.png">
this is the current behaviour, in which user are told in a dialog box that they are deleting a specific Virtual Machine.(FreeNASRC2 in this case)

Reply:
@wblock let me know if this is desired behaviour